test/index.js

```diff
@@ -233,18 +237,94 @@ describe("loader-utils", function() {
 		]);
 	});
 
-	// request with relative path args
 	describe("#stringifyRequest", function() {
-		var pathArgsString = "path=../../thing/file";
-		var pathArgsObjString = JSON.stringify( { "path-to-file": "../../thing/file" } );
-		[
-			["/path/to/thing", "/path/to/thing/file!/path/to/thing/file?" + pathArgsString, JSON.stringify("./file!./file?" + pathArgsString)],
-			["/path/to/thing", "/path/to/thing/file!/path/to/thing/file?" + pathArgsObjString, JSON.stringify("./file!./file?" + pathArgsObjString)]
-		].forEach(function(test) {
-			it("should stringify " + test[1] + " " + test[2], function() {
-				var request = loaderUtils.stringifyRequest({ context: test[0] }, test[1]);
-				assert.equal(request, test[2]);
-			});
+		// We know that query strings that contain paths and question marks can be problematic.
+		// We must ensure that stringifyRequest is not messing with them
+		var paramQueryString = "?questionMark?posix=path/to/thing&win=path\\to\\thing";
+		var jsonQueryString = "?" + s({
+			questionMark: "?",
+			posix: "path/to/thing",
+			win: "path\\to\\file"
+		});
+		var win = path.sep === "\\";
+		var posix = path.sep === "/";
+		var cases = [
+			{ request: "./a.js", expected: s("./a.js") },
+			{ request: ".\\a.js", expected: s("./a.js") },
+			{ request: "./a/b.js", expected: s("./a/b.js") },
+			{ request: ".\\a\\b.js", expected: s("./a/b.js") },
+			{ request: "module", expected: s("module") }, // without ./ is a request into the modules directory
+			{ request: "module/a.js", expected: s("module/a.js") },
+			{ request: "module\\a.js", expected: s("module/a.js") },
+			{ request: "./a.js" + paramQueryString, expected: s("./a.js" + paramQueryString) },
+			{ request: "./a.js" + jsonQueryString, expected: s("./a.js" + jsonQueryString) },
+			{ request: "module" + paramQueryString, expected: s("module" + paramQueryString) },
+			{ request: "module" + jsonQueryString, expected: s("module" + jsonQueryString) },
+			posix && { context: "/path/to", request: "/path/to/module/a.js", expected: s("./module/a.js") },
+			win && { context: "C:\\path\\to\\", request: "C:\\path\\to\\module\\a.js", expected: s("./module/a.js") },
+			posix && { context: "/path/to/thing", request: "/path/to/module/a.js", expected: s("../module/a.js") },
+			win && { context: "C:\\path\\to\\thing", request: "C:\\path\\to\\module\\a.js", expected: s("../module/a.js") },
+			// If context and request are on different hard drives, the path should not be relative
+			// @see https://github.com/webpack/loader-utils/pull/14
+			win && { context: "D:\\path\\to\\thing", request: "C:\\path\\to\\module\\a.js", expected: s("C:/path/to/module/a.js") },
```

Comment:
I think, it would make sense to make `/` as the only separator for requests. Makes it easier for the future. And `/` are also valid on Windows.

What do you think?